src/kokkos/test/kokkos/testEigenGPU.cc

```diff
@@ -0,0 +1,426 @@
+  if (N > 3)
+    hits_ge.col(3)[1] = -5.28e-06;
+}
+
+template <int N>
+KOKKOS_INLINE_FUNCTION void kernelFillHitsAndHitsCov(Kokkos::View<double*, KokkosExecSpace> vhits,
+                                                     Kokkos::View<float*, KokkosExecSpace> vhits_ge,
+                                                     const member_type& teamMember) {
+  auto i = teamMember.league_rank() * teamMember.team_size() + teamMember.team_rank();
+
+  double* __restrict__ phits = vhits.data();
+  float* __restrict__ phits_ge = vhits_ge.data();
+
+  Rfit::Map3xNd<N> hits(phits + i, 3, N);
+  Rfit::Map6xNf<N> hits_ge(phits_ge + i, 6, N);
+  hits_ge = MatrixXf::Zero(6, N);
+  fillHitsAndHitsCov(hits, hits_ge);
+}
+
+template <int N>
+void testFit() {
+  constexpr double B = 0.0113921;
+
+  Kokkos::View<double*, KokkosExecSpace> d_hits("d_hits", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix3xNd<N>));
+  Kokkos::View<float*, KokkosExecSpace> d_hits_ge("d_hits_ge", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix6xNf<N>));
+  Kokkos::View<double*, KokkosExecSpace> d_fast_fit_results("d_fast_fit_results",
+                                                            Rfit::maxNumberOfTracks() * sizeof(Vector4d));
+  Kokkos::View<Rfit::line_fit*, KokkosExecSpace> d_line_fit_results("d_line_fit_results",
+                                                                    Rfit::maxNumberOfTracks() * sizeof(Rfit::line_fit));
+  Kokkos::View<Rfit::circle_fit*, KokkosExecSpace> d_circle_fit_results(
+      "d_circle_fit_results", Rfit::maxNumberOfTracks() * sizeof(Rfit::circle_fit));
+
+  Rfit::Matrix3xNd<N> hits;
+  Rfit::Matrix6xNf<N> hits_ge = MatrixXf::Zero(6, N);
+
+  double* fast_fit_resultsGPUret = new double[Rfit::maxNumberOfTracks() * sizeof(Vector4d)];
+  Rfit::circle_fit* circle_fit_resultsGPUret = new Rfit::circle_fit();
+  Rfit::line_fit* line_fit_resultsGPUret = new Rfit::line_fit();
+
+  fillHitsAndHitsCov(hits, hits_ge);
+
+  std::cout << "sizes " << N << ' ' << sizeof(hits) << ' ' << sizeof(hits_ge) << ' ' << sizeof(Vector4d) << ' '
+            << sizeof(Rfit::line_fit) << ' ' << sizeof(Rfit::circle_fit) << std::endl;
+
+  std::cout << "Generated hits:\n" << hits << std::endl;
+  std::cout << "Generated cov:\n" << hits_ge << std::endl;
+
+  // FAST_FIT_CPU
+  Vector4d fast_fit_results;
+#ifdef USE_BL
+  BrokenLine::BL_Fast_fit(hits, fast_fit_results);
+#else
+  Rfit::Fast_fit(hits, fast_fit_results);
+#endif
+  std::cout << "Fitted values (FastFit, [X0, Y0, R, tan(theta)]):\n" << fast_fit_results << std::endl;
+
+  // cudaMemset d_fast_fit_results & d_line_fit_results to 0
+  int tmpN = Rfit::maxNumberOfTracks() * sizeof(Vector4d);
+  Kokkos::parallel_for(
+      "init_fast_hit_res",
+      Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(), 0, tmpN),
+      KOKKOS_LAMBDA(const int& i) {
+        unsigned char* ptr = reinterpret_cast<unsigned char*>(d_fast_fit_results.data());
+        ptr[i] = 0;
+      });
```

Comment:
Why not just `Kokkos::deep_copy(KokkosExecSpace(), d_fast_fit_results, 0)`?

Reply:
Works for `d_fast_fit_results` but not for `d_line_fit_results` so I chose for-loops for consistency.

Reply:
Ok. Could `d_line_fit_results` be done by explicitly assigning zeros instead of the `reinterpter_cast`? I'm thinking something along
```
d_line_fit_results[i].par = Vector2d::Zero(); 
d_line_fit_results[i].cov = Matrix2d::Zero();
d_line_fit_results[i].chi2 = 0.;
```
I think the intention would be more clear.